chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/auditdata/LoggedActionsAuditDataRetentionService.java

```diff
@@ -0,0 +1,82 @@
+package gov.healthit.chpl.scheduler.job.auditdata;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.sql.SQLException;
+
+import javax.persistence.Query;
+import javax.transaction.Transactional;
+
+import org.postgresql.copy.CopyManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@Component
+public class LoggedActionsAuditDataRetentionService  extends AuditDataRetentionService {
+    private static final String AUDIT_NAME = "audit.logged_actions";
+
+    @Autowired
+    public LoggedActionsAuditDataRetentionService(@Value("${auditDataFilePath}") String auditDataFilePath) {
+        super(auditDataFilePath);
+    }
+
+    @Override
+    public Long getAuditDataCount(Integer month, Integer year) {
+        Query query = entityManager.createQuery(
+                "SELECT count(*) "
+                + "FROM LoggedActionEntity a "
+                + "WHERE MONTH(a.actionTStamp) = :month "
+                + "AND YEAR(a.actionTStamp) = :year");
+        query.setParameter("month", month);
+        query.setParameter("year", year);
+
+        return (Long) query.getSingleResult();
+    }
+
+    @Override
+    @Transactional
+    public void deleteAuditData(Integer month, Integer year) {
+        Query query = entityManager.createQuery(
+                "DELETE FROM LoggedActionEntity a "
+                + "WHERE MONTH(a.actionTStamp) = :month "
+                + "AND YEAR(a.actionTStamp) = :year");
+        query.setParameter("month", month);
+        query.setParameter("year", year);
+
+        query.executeUpdate();
+    }
+
+    @Override
+    public void archiveDataToFile(Integer month, Integer year, String fileName, boolean includeHeaders) throws SQLException {
+        String copyCmd = "COPY "
+                + "(SELECT * "
+                + "FROM audit.logged_actions "
+                + "WHERE EXTRACT(MONTH FROM action_tstamp) = " + month.toString() + " "
```

Comment:
Same question I guess about if this is a safe way to create the SQL. If you have an entity I think you can declare a native sql query in that. Might be another idea.. I've done it in some of the Activity entities.